dissect/target/plugins/os/windows/syscache.py

```diff
@@ -75,7 +75,9 @@ def syscache(self):
                 full_path = None
                 if mft:
                     try:
-                        full_path = self.target.fs.path("\\".join(["sysvol", mft.mft(file_segment).fullpath()]))
+                        path  = mft.ntfs.mft(file_segment).full_path()
```

Comment:
Looks good, thanks!

That was first time running syscache plugin for me. My guess about regression in ``dissect.nfts`` was based on what looked like a rename in API function that was missed here when updating another parts of dissect modules. Not sure whether tests for syscache are working, maybe that's why it was missed.

Also I did some more debbuging - this happened when triaging virtualized Server 2008 that was hit with ransomware. When running this plugin it fails to get full paths from MFT records for 100+ records, no clue why. Some info from debugger for one file, there is no inode ``75390`` in MFT - either this is a quirk that came from MFT changes on the system, or perhaps a bug in ``dissect.ntfs``:

```
ipdb> rec = mft(file_segment)
ipdb> rec
<MftRecord 75390#902>
ipdb> dir(rec)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_get_stream_attributes', 'attributes', 'data', 'dataruns', 'filename', 'filenames', 'from_bytes', 'from_fh', 'full_path', 'full_paths', 'get', 'has_stream', 'header', 'index', 'is_dir', 'is_file', 'is_mount_point', 'is_reparse_point', 'is_symlink', 'iterdir', 'listdir', 'ntfs', 'offset', 'open', 'reparse_point_name', 'reparse_point_record', 'reparse_point_substitute_name', 'resident', 'segment', 'size']
ipdb> rec.full_path()
ipdb> rec.filenames
<bound method MftRecord.filenames of <MftRecord 75390#902>>
ipdb> rec.filenames()
[]
ipdb> rec.data
b'FILE0\x00\x03\x00v\xed\x11A7\x00\x00\x00\x86\x03\x00\x008\x00\x01\x00\xf8\x03\x00\x00\x00\x04\x00\x00\x1d&\x01\x00\x00\x00c\x02\x01\x00\x00\x00~&\x01\x00\x06\x00!\x10\x00\x00\x00\x00\x80\x00\x00\x00\xb8\x03\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xef\r\x00\x00\x00\x00\x00\x00@\x00\x00\x00\x00\x00\x00\x00\x00\x10\xf0\x14\x00\x00\x00\x00\x00\x03\xf0\x14\x00\x00\x00\x00\x00\x03\xf0\x14\x00\x00\x00\x001\x01d\'21\x03\x05\xfd\xd11\x04M%\n1\x08luo!\x10\xab(!\x10\xe3\x1e!\x10\xd2\x001\x10\xc39\xff!\x10D\x93!\x10\x99\xdc!\x101\xea!\x04\xd3\xf9!\x01\xbb\xf91\x03\xe5q#1\x04c\xd2+1\x0c\xef\x99\xf7A\x19\x97\xcfQ\x011\x0f\xba\x1c\xe8A\x10\x85\xf0\xee\xfe1\x10\xba\x16\xe1!\x10\n\xaa1\x10\xfc\xf4\xf3\x11\x10\xeb!\x10>\xd3\x11\x10\xdf!\x10\x9f\xde!\x10\r\xfd!\x10\xff\xc6!\x10P\xff\x11\x10\xdd!\x10\x89\xa31\x10\xe6I\xfe1\x10\x078\xff\x11\x10\xb1!\x10R\xff1\x10d\x9e\xf91\x10e\xfc\xfb1\x10\xaf\x82\xfb1\x10\x1bq\xfe1\x10:j\xff1\x10k-\xfe1\x10\x8a\x15\xfd!\x10\x00\xff\x11\x10\xc0!\x10 \xf9\x11\x10\xa0\x11\x10\xc0\x11\x10\xc0!\x10`\xff\x11\x10\xb0\x11\x10\x90!\x10H\xff!\x10r\xff!\x10\x86\xf8\x11\x10\xc0\x11\x10\xc0!\x10P\xff!\x10\x00\xff\x11\x10\xb0!\x10\xff!\x10\x92\xfc!\x10\xc0\xf51\x10\xe6\xd8\xfe\x11\x10\xc8\x11\x10\xb9\x11\x10\xc2!\x10W\xff!\x10u\xb0!\x10\xb0\xfa\x11\x10\xe0\x11\x10\xe0\x11\x10\xe0\x11\x10\xa0\x11\x10\xc0!\x10\x86\xf1\x11\x10\xe0\x11\x10\xe0!\x10\xce\x911\x10\xe9P\xfd!\x10\xb4\xb0!\x10\xb2\xf91\x10\xb5C\xfd\x11\x10\x8e1\x10ZH\xfb1\x10Xe\xfe1\x10\xafK\xff!\x10\xc2\xbe!\x10A\xfd!\x10\xe9\xf8!\x10\xb0\xae\x11\x10\xe0\x11\x10\xe0!\x10\x12\xfd\x11\x10\xe0!\x10\xda\xf5!\x10#\xf9\x11\x10\xe0!\x10\x85\xfd\x11\x10\xe0!\x10\xd5\xed1\x10\x9f\xe5\xfe1\x10\xd5k\xfe!\x10X\xf9\x11\x10\xaa\x11\x10\xe7\x11\x10\xd5!\x10\xdb\xb41\x10\xc03\xfb1\x10\xc1\xf9\xfe!\x10\xfd\xe4!\x10U\xbc\x11\x10\xe0!\x10.\xb4\x11\x10\xe6!\x10\x98\xc2!\x10{\xf71\x10\xf9\x83\xfe1\x10n\x7f\xfe!\x10T\x9f\x11\x10\xe0!\x10n\xfa!\x103\xef1\x10\xb2g\xff!\x10\xe5\xfc\x11\x10\xe0\x11\x10\xc0\x11\x10\xe4!\x10O\xfa!\x10\xc9\xfb!\x10/\xe2!\x10\xb2\xfe\x11\x10\xb5\x11\x10\xe0\x11\x10\xe0!\x10\xc5\xf6!\x10.\xf6\x11\x10\xe0\x11\x10\xe0!\x10\xa2\xfc!\x10R\xff!\x10\x11\xfc!\x10Y\xd1!\x10\xc8\xe2!\x10\xdb\xd7!\x10[\xd7!\x10\x12\xc6!\x10\xce\xe7!\x10#\xd9!\x10\xcc\xf2!\x10o\xf4\x11\x10\xe4!\x10\x05\xbf!\x10D\xf0!\x10\xef\xfb1\x10\xdc\x1a\xff!\x10i\xe3!\x10\xf0\xdf!\x10e\xce\x11\x10\xb5!\x10g\xd9\x11\x10\xc7!\x10\xd3\xf5!\x104\xfe!\x10\x85\xf7\x11\x10\xe0\x11\x10\xe3!\x10\x06\xe9!\x105\xcb\x11\x10\xe8!\x10\xf6\xfa!\x10\x1d\xff\x11\x10\xe6!\x10\xc7\xf3\x11\x10\xe7\x11\x10\xec!\x10$\xff\x11\x10\xac\x11\x10\xd7!\x10\x8c\x96\x11\x10\xe9!\x10\xe6\xf0\x11\x10\xeb\x11\x10\xc3!\x10\x82\xfd\x11\x10\xe8\x11\x10\xde!\x10\xbb\xe3!\x10\xf8\xd7!\x10!\x86!\x10\x11\xac\x11\x10\xe8!\x10y\xfc1\x10\xb2n\xfe!\x10y\xef\x11\x10\xa8!\x10\xe5\xfe!\x10\x89\xf0!\x10\x08\xff!\x10\xc5\xf8!\x10\xc0\xd9\x11\x10\xe6!\x10\xdb\xf0!\x10\xa3\xfb!\x10d\xf7\x11\x10\xe9!\x10[\xf5\x11\x10\xe6!\x10"\xff!\x10\xa2\xf5\x11\x10\x88!\x10T\xfa\x11\x10\xe0!\x10\xe7\xf8!\x10\x8a\xf2\x11\x10\xe8!\x10\xfd\xf6!\x10\xa7\xf6!\x10{\xfc!\x10\xd8\xa1\x11\x10\xe8!\x10;\xf8!\x10$\xe51\x10\xf6,\xfe!\x10x\xf9!\x10\xe1\x8f1\x10\t\x80\xfc\x11\x10\xe0\x11\x10\xe2!\x10v\xef\x00\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
```

I'm not even sure what's better - to just skip such syscache entries or return empty ``full_path`` (since another data in syscache entry can be based on wrong MFT information)

Reply:
> My guess about regression in dissect.nfts was based on what looked like a rename in API function that was missed here when updating another parts of dissect modules. Not sure whether tests for syscache are working, maybe that's why it was missed.

That would be correct. `dissect.ntfs` was rewritten from scratch just before our open-source release, and it looks like we missed this plugin.

Reply:
> Also I did some more debbuging - this happened when triaging virtualized Server 2008 that was hit with ransomware. When running this plugin it fails to get full paths from MFT records for 100+ records, no clue why. Some info from debugger for one file, there is no inode 75390 in MFT - either this is a quirk that came from MFT changes on the system, or perhaps a bug in dissect.ntfs:

Not sure why it cannot find the full paths for those records. Can you verify if the attributes are present in the raw record data? If so, you may create a new issue on  `dissect.ntfs`. At least we got the `AttributeError` fixed on this PR.